### PR TITLE
Removing mesh

### PR DESCRIPTION
**Pull request by hjabbot**
# PolarRoute Pull Request Template

Date: 2023-09-19
Version Number: 0.3.4
## Description of change
Removing the mesh construction from PolarRoute as it has been put into its [own repo](https://github.com/antarctica/CartograPhi). 

## Fixes  (https://github.com/antarctica/PolarRoute/issues/146)

# Testing
To ensure that the functionality of the PolarRoute codebase remains consistent throughout the development cycle a testing strategy has been developed, which can be viewed in the document `test/testing_strategy.md`. 
This includes a collection of test files which should be run according to which part of the codebase has been altered in a pull request. Please consult the testing strategy to determine which tests need to be run. 

- [ ] My changes have not altered any of the files listed in the testing strategy

- [ ] My changes result in all required regression tests passing without the need to update test files.  

- [x] My changes require one or more test files to be updated for all regression tests to pass.   

> Moved tests into new repo, [here](https://github.com/antarctica/PolarRoute/files/12662331/reg_test_pr.txt) are all the remaining tests passing 

# Checklist

- [x] My code follows [pep8](https://peps.python.org/pep-0008/) style guidelines.  
- [x] I have commented my code, particularly in hard-to-understand areas.  
- [x] I have updated the documentation of the codebase where required.  
- [x] My changes generate no new warnings.   
- [x] My PR has been made to the `0.3.x` branch (**DO NOT SUBMIT A PR TO MAIN**)

### CODE REVIEW

File: polar_route/vessel_performance/vessels/underwater_vessel.py
Comment (gecoombs) [inline polar_route/vessel_performance/vessels/underwater_vessel.py:1]: I don't think we need this file anymore, it has been replaced by the glider models and any other underwater vehicles will probably require their own classes.

File: docs/source/index.rst
Comment (gecoombs) [inline docs/source/index.rst:1]: Probably worth mentioning here that PolarRoute depends on Cartographi and maybe that docs for the mesh generation can be found on that repo?

**Review by SamuelHall700 — CHANGES_REQUESTED** [review]
All tests passed, and installation into a fresh venv also worked - only minor changes were requested. Good work!

**Review by ayat-khairy — APPROVED** [review]
Looks good. We just need to clean up the utils to make sure that they are not duplicated across the 2 repos at some point.

**Review by gecoombs — CHANGES_REQUESTED** [review]
Looks good, happy to approve if you just delete `underwater_vessel.py` and maybe reference the mesh repo in the intro to the docs.

**Pull request merged**